pkg/scheduler/internal/cache/cache.go

```diff
@@ -435,18 +428,23 @@ func (cache *schedulerCache) updatePod(oldPod, newPod *v1.Pod) error {
 }
 
 // Assumes that lock is already acquired.
-// Removes a pod from the cached node info. When a node is removed, some pod
-// deletion events might arrive later. This is not a problem, as the pods in
-// the node are assumed to be removed already.
+// Removes a pod from the cached node info. If the node information was already
+// removed and there are no more pods left in the node, cleans up the node from
+// the cache.
 func (cache *schedulerCache) removePod(pod *v1.Pod) error {
 	n, ok := cache.nodes[pod.Spec.NodeName]
 	if !ok {
+		klog.Errorf("node %v not found when trying to remove pod %v", pod.Spec.NodeName, pod.Name)
```

Comment:
I recalled the original logic returned an error?

Reply:
It did. But returning nil is actually safer in the case of extraneous update events that might arrive before a node is created, and after the original node was completely removed.

Reply:
+1 to returning nil and just logging an error.

Reply:
I checked the usage of `removePod()`, there are still a number of callers rely on the returned value. So I'd suggest to revert to the original state.

Reply:
Detail for each caller:

- ForgetPod: We actually want to proceed and clear the assumedPods and podStates.
- expirePod: Same as above.
- AddPod: it just logs the error returned, so same effect.
- RemovePod: We want to clear podStates.
- updatePod: This is the case where we want to prevent losing information.

Reply:
That said, for expirePod and ForgetPod, the node shouldn't have been removed because pods were already assigned.

Reply:
Thanks. That's fair.